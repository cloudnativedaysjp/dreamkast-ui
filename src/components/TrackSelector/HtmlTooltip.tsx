import { Theme, Tooltip } from '@material-ui/core'
import { withStyles } from '@material-ui/styles'

const HtmlTooltip = withStyles((theme: Theme) => ({
  tooltip: {
    backgroundColor: '#f5f5f9',
    color: 'rgba(0, 0, 0, 0.87)',
    maxWidth: 220,
    fontSize: theme.typography.pxToRem(12),
    border: '1px solid #dadde9',
  },
}))(Tooltip)

export default HtmlTooltip
