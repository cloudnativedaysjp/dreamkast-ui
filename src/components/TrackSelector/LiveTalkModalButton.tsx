import React, { ReactElement, useState } from 'react'
import * as Styled from './styled'
import { FormatListBulleted } from '@material-ui/icons'

type Props = {
  content: (closeModal: () => void) => ReactElement
}

export const LiveTalkModalButton: React.FC<Props> = ({ content }) => {
  const [isModalOpen, setModalOpen] = useState<boolean>(false)

  return (
    <>
      <Styled.LiveTalkModalButton
        data-testid="btn"
        color="primary"
        onClick={() => setModalOpen(true)}
      >
        <FormatListBulleted />
      </Styled.LiveTalkModalButton>
      <Styled.LiveTalkModal
        open={isModalOpen}
        onClose={() => setModalOpen(false)}
      >
        {content(() => setModalOpen(false))}
      </Styled.LiveTalkModal>
    </>
  )
}
