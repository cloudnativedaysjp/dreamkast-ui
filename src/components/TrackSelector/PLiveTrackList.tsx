import React from 'react'
import * as Styled from './styled'
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableRow,
} from '@material-ui/core'
import dayjs from 'dayjs'
import { Talk, Track } from '../../generated/dreamkast-api.generated'

type Props = {
  data: { track: Track; talk?: Talk }[]

  selectedTrack: number

  onChange: (selectedItem: number | null) => void
}

export const PLiveTrackList: React.FC<Props> = ({
  data,
  selectedTrack,
  onChange,
}) => {
  return (
    <Styled.Container>
      <Styled.InnerContainer>
        <Styled.Title>ライブセッション</Styled.Title>
        <Styled.List>
          <TableContainer>
            <Table aria-label="live-talks">
              <TableBody>
                {data.map(({ talk, track }) => (
                  <TableRow key={track.id}>
                    <TableCell style={{ borderBottom: 'none', width: '150px' }}>
                      <Styled.TrackSelectorButton
                        className={track.id === selectedTrack ? 'selected' : ''}
                        onClick={() => onChange(track.id)}
                      >
                        {track.name}
                      </Styled.TrackSelectorButton>
                    </TableCell>
                    <TableCell align={'left'} style={{ borderBottom: 'none' }}>
                      {talk && (
                        <>
                          {talk.onAir && <Styled.Live>LIVE</Styled.Live>}{' '}
                          {dayjs(talk.startTime).format('HH:mm')}-
                          {dayjs(talk.endTime).format('HH:mm')}
                          <br />
                          {talk.title}
                          <br />
                          {talk.speakers?.map((s) => s.name).join(', ')}
                        </>
                      )}
                    </TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
        </Styled.List>
      </Styled.InnerContainer>
    </Styled.Container>
  )
}
