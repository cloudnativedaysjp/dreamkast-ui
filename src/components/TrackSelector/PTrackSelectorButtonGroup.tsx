import React from 'react'
import * as Styled from './styled'
import { Talk, Track } from '../../generated/dreamkast-api.generated'
import { Theme, Tooltip } from '@material-ui/core'
import dayjs from 'dayjs'
import { withStyles } from '@material-ui/styles'

type Props = {
  data: { track: Track; talk?: Talk }[]
  selectedTrack: number
  onChange: (selectedItem: number | null) => void
}

export const PTrackSelectorButtonGroup: React.FC<Props> = ({
  data,
  selectedTrack,
  onChange,
}) => {
  return (
    <Styled.TrackSelectorButtonGroup
      value={selectedTrack}
      color="primary"
      onChange={(_, i) => onChange(i)}
      exclusive
    >
      {data.map(({ track, talk }) => (
        <HtmlTooltip
          key={track.id}
          title={
            <React.Fragment>
              <div>
                {!talk ? (
                  'ライブセッションはありません'
                ) : (
                  <>
                    {talk.onAir && <Styled.Live>LIVE</Styled.Live>}{' '}
                    {dayjs(talk.startTime).format('HH:mm')}-
                    {dayjs(talk.endTime).format('HH:mm')}
                    <br />
                    {talk.title}
                    <br />
                    {talk.speakers?.map((s) => s.name).join(', ')}
                  </>
                )}
              </div>
            </React.Fragment>
          }
        >
          <Styled.MenuItem key={track.id} value={track.id}>
            {track.name}
          </Styled.MenuItem>
        </HtmlTooltip>
      ))}
    </Styled.TrackSelectorButtonGroup>
  )
}

const HtmlTooltip = withStyles((theme: Theme) => ({
  tooltip: {
    backgroundColor: '#f5f5f9',
    color: 'rgba(0, 0, 0, 0.87)',
    maxWidth: 220,
    fontSize: theme.typography.pxToRem(12),
    border: '1px solid #dadde9',
  },
}))(Tooltip)
