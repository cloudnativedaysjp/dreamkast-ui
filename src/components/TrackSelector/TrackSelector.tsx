import React, { useState, useEffect } from 'react'
import * as Styled from './styled'
import { Track } from '../../client-axios'

type Props = {
  tracks: Track[]
  selectedTrackId: number
  selectTrack: (selectedId: number) => void
}

export const TrackSelector: React.FC<Props> = ({
  tracks,
  selectedTrackId,
  selectTrack,
}) => {
  const [item, setItem] = useState(selectedTrackId)

  useEffect(() => {
    setItem(selectedTrackId)
  }, [selectedTrackId])

  const handleChange = (
    _event: React.MouseEvent<HTMLElement>,
    selectItem: number | null,
  ) => {
    if (selectItem !== null) {
      setItem(selectItem)
      selectTrack(selectItem as number)
    }
  }
  return (
    <Styled.TrackMenuContainer
      value={item}
      color="primary"
      onChange={handleChange}
      exclusive
    >
      {tracks.map((track) => (
        <Styled.MenuItem key={track.id} value={track.id}>
          {track.name}
        </Styled.MenuItem>
      ))}
    </Styled.TrackMenuContainer>
  )
}
