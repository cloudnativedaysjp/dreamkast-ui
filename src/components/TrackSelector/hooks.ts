import { useSelector } from 'react-redux'
import { settingsSelector } from '../../store/settings'
import { Track } from '../../generated/dreamkast-api.generated'

export const useTracksWithLiveTalk = (tracks: Track[]) => {
  const { talks } = useSelector(settingsSelector)

  const data = tracks.map((tr) => {
    const talkId = tr.onAirTalk
      ? ((tr.onAirTalk as any)?.talk_id as string)
      : ''
    return {
      track: tr,
      talk: talks[talkId],
    }
  })

  return data
}
