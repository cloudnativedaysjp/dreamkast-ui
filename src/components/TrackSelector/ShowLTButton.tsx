import React from 'react'
import * as Styled from './styled'
import { PeopleAlt } from '@material-ui/icons'
import { showLT } from '../../store/settings'
import { useDispatch } from 'react-redux'
import HtmlTooltip from './HtmlTooltip'

export const ShowLTButton: React.FC = () => {
  const dispatch = useDispatch()

  return (
    <HtmlTooltip title="コミュニティLT">
      <Styled.ShowLTButton
        data-testid="btn"
        color="primary"
        onClick={() => {
          dispatch(showLT())
        }}
      >
        <PeopleAlt />
      </Styled.ShowLTButton>
    </HtmlTooltip>
  )
}
