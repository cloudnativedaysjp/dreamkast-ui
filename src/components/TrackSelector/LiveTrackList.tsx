import React from 'react'
import * as Styled from './styled'
import { Track } from '../../generated/dreamkast-api.generated'
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableRow,
} from '@material-ui/core'
import dayjs from 'dayjs'
import { useTracksWithLiveTalk } from './hooks'

type Props = {
  selectedTrack: number
  tracks: Track[]

  onChange: (selectedItem: number | null) => void
}

export const LiveTrackList: React.FC<Props> = ({
  selectedTrack,
  tracks,
  onChange,
}) => {
  const data = useTracksWithLiveTalk(tracks)

  return (
    <Styled.Container>
      <Styled.InnerContainer>
        <Styled.Title>ライブセッション</Styled.Title>
        <Styled.List>
          <TableContainer>
            <Table aria-label="live-talks">
              <TableBody>
                {data.map(({ talk, track }) => (
                  <TableRow key={track.id}>
                    <TableCell style={{ borderBottom: 'none', width: '150px' }}>
                      <Styled.TrackSelectorButton
                        className={track.id === selectedTrack ? 'selected' : ''}
                        onClick={() => onChange(track.id)}
                      >
                        {track.name}
                      </Styled.TrackSelectorButton>
                    </TableCell>
                    <TableCell align={'left'} style={{ borderBottom: 'none' }}>
                      {!talk ? (
                        ''
                      ) : (
                        <>
                          {talk.onAir && <Styled.Live>LIVE</Styled.Live>}{' '}
                          {dayjs(talk.startTime).format('HH:mm')}-
                          {dayjs(talk.endTime).format('HH:mm')}
                          <br />
                          {talk.title}
                          <br />
                          {talk.speakers?.map((s) => s.name).join(', ')}
                        </>
                      )}
                    </TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
        </Styled.List>
      </Styled.InnerContainer>
    </Styled.Container>
  )
}
