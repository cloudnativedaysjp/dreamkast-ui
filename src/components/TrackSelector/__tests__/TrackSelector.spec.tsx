import React from 'react'
import renderer from 'react-test-renderer'
import { TrackSelector } from '../TrackSelector'
import { MockTracks } from '../../../testhelper/fixture'

test.skip('TrackSelector', () => {
  const component = renderer.create(
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    <TrackSelector
      selectedTrack={MockTracks()[0]}
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      selectTrack={() => {}}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
