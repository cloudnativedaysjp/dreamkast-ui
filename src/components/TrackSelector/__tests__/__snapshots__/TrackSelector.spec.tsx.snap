// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTrackSelector should render child component 1`] = `
<DocumentFragment>
  <div
    data-testid="tgt"
  />
</DocumentFragment>
`;
